### ISSUE TITLE
sshd filter does detect public key failure

### ISSUE THREAD

**Issue by llamafilm** [fail2ban/fail2ban#2440]
On all my servers, I like to disable SSH PasswordAuthentication.  This means that failed SSH logins look like this in the log:

```
Jun  5 11:09:40 arsenal sshd[32595]: Connection closed by 192.168.1.202 port 61745 [preauth]
```

The default sshd filter does not detect this:

```
2019-06-05 11:32:49,837 fail2ban.filtersystemd  [33604]: DEBUG   Read systemd journal entry: '2019-06-05T11:32:49.837498arsenal sshd[33846]: Connection closed by 192.168.1.202 port 61965 [preauth]'
```

Even though the public key is secure, I would still like to use fail2ban to drop the packets.  Currently, the attacher gets this info:

```
root@192.168.1.1: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
```

**Comment by sebres** [comment]
> On all my servers, I like to disable SSH PasswordAuthentication. This means that failed SSH logins look like this in the log...
sshd[32595]: Connection closed by 192.168.1.202 port 61745 [preauth]

Nope, this means that the message with `sshd[32595]` is never alone if an authentication would take place at all (you should also see another messages too)...
And failed ssh logins with public keys are normally logged as something like this:
https://github.com/fail2ban/fail2ban/blob/ec681a336350f193f12d6c4c307b7355bd3770ca/fail2ban/tests/files/logs/sshd#L168-L172
And our current filter does detect that (at least the end of unsuccessful authentication phase).

> The default sshd filter does not detect this...

People, we have made our issue template not for nothing. 
Which fail2ban version it is?
Please provide also a journal excerpt of **all messages** with `sshd[32595]`...

Actually "Connection closed by ... [preauth]" is not an authentication issue, but fail2ban uses a matching rule as helper to identify IP address for other failures with the same session-id (multiline failure-ID `<F-MLFID>`). but it could be activated as failure using mode `ddos` or `aggressive` (see https://github.com/fail2ban/fail2ban/issues/2361#issuecomment-479703314).

For more info see also #1719 and others (just search for issues with `Connection closed by`).

**Comment by llamafilm** [comment]
Alright, sorry for the bad report.  I glanced at the latest branch and didn't see any differences in the sshd filter so I figured the version didn't matter.  I'm using version 0.9.7 installed through yum on CentOS 7.

I had sshd LogLevel = INFO (the default setting).  After changing it to VERBOSE, now I see the additional line with "Failed publickey" and fail2ban is working.
